test/internet/test-dns-lookup.js

```diff
@@ -28,3 +30,17 @@ assert.rejects(
     message: `getaddrinfo ENOTFOUND ${addresses.INVALID_HOST}`
   }
 );
+
+dns.lookup(addresses.INVALID_HOST, {
+  hints: 0,
+  family: 0,
+  all: true
+}, common.mustCall((error) => {
+  assert.strictEqual(error.code, 'ENOTFOUND');
+  assert.strictEqual(
+    error.message,
+    `getaddrinfo ENOTFOUND ${addresses.INVALID_HOST}`
+  );
+  assert.strictEqual(error.syscall, 'getaddrinfo');
+  assert.strictEqual(error.hostname, addresses.INVALID_HOST);
+}));
```

Comment:
As a hint: it would be easier to write this as:

```js
assert.throws(
  () => dns.lookup(addresses.INVALID_HOST, {
    hints: 0,
    family: 0,
    all: true
  }),
  {
    code: 'ENOTFOUND',
    message: 'getaddrinfo ENOTFOUND ...',
    syscall: 'getaddrinfo',
    hostname: addresses.INVALID_HOST
  }
);